lib/npm-provider.js

```diff
@@ -36,6 +36,15 @@ function mapNpmDownloads(camp, urlComponent, apiUriComponent) {
   }));
 }
 
+// npm registry url
+function getNpmRegistryUrl(maybeProtocol, maybeHost) {
```

Comment:
Using the query string should make this a little simpler. If you still need this function it should go in a new file `npm-badge-helpers.js`.

Reply:
what is the difference between `npm-provider` and `npm-badge-helpers`?

Reply:
Most of the badge providers are in `server.js`, though a few, like npm, are pulled out. We use `-provider` to refer to the function which implements the badge.

The `-badge-helpers` are service-specific utility functions which are invoked by the providers.

Yea, in a couple cases the line is gray. Though this one seems to be a utility function.

Reply:
moved it

Reply:
and i got rid off of helper function, now i only use default npm registry uri as a constant to share